pkg/controller/seed-controller-manager/mla/datasource_grafana_controller.go

```diff
@@ -144,10 +148,22 @@ func (r *datasourceGrafanaReconciler) reconcile(ctx context.Context, cluster *ku
 	if cluster.Spec.MLA == nil {
 		cluster.Spec.MLA = &kubermaticv1.MLASettings{}
 	}
+	grafanaClient := grafanasdk.NewClient(r.grafanaURL, r.grafanaAuth, r.httpClient)
```

Comment:
So this means in every reconcilation loop, you will create a new grafana client.
Is this what we really want to do?
I am thinking that we could still pass the grafanaclient to this reconciler, and only set header in the reconciliation loop.
WDYT?

Reply:
yeah, that was my first idea, but in this way, we will have the same race condition, cause we are using one client for all workers.
creating a client does not do much, it's just creating structure, so it doesn't cost much for us.